### DISCUSSION TITLE
kafka connector  can  retrieve secret from azure key vault

### DISCUSSION THREAD

**Discussion by chanvoyage** [strimzi/strimzi-kafka-operator#5410] [Q&A]
hello guy

Is possible to get credentials directly azure key vault to kafka connector

example:

Currently my password store on azure key vault  , When use key vault  I will use cmd "az key vault  get credential  and replace by
cmd "sed -i 's|password-key01|${passowrd1234}' path/to/source.yaml "

it does not look secured !  Do you have any suggestion  for retrieve secret ?  

![image](https://user-images.githubusercontent.com/39899497/128806757-d46310a6-b312-45e8-89cd-7022416d7bde.png)

 

thank you

**Comment by scholzj** [comment]
Strimzi currently supports reading external values from Kubernetes Secrets and ConfigMaps. There is no support for Azure Key Vault. If there is some Kafka Configuration Provider for it, you can probably add it to the Connect image and use it. Otherwise you would probably need to map it first to Kubernetes Secret and in some way and then use the Secret.
